packages/components/src/ErrorPage/ErrorPage.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { HTMLAttributes } from "react"
+import { FormattedMessage, useIntl } from "@cultureamp/i18n-react-intl"
+import classNames from "classnames"
+import { BrandMomentError } from "@kaizen/draft-illustration"
+import { Paragraph } from "@kaizen/typography"
+import { BrandMoment } from "~components/BrandMoment"
+import { ArrowRightIcon } from "~components/SVG/icons/ArrowRightIcon"
+import { EmailIcon } from "~components/SVG/icons/EmailIcon"
+import { OverrideClassName } from "~types/OverrideClassName"
+import { useMessages } from "./hooks"
+import styles from "./ErrorPage.module.scss"
+
+const getMailToHref = (code: number): string => {
+  const supportEmail = "support@cultureamp.com"
+  const subject = "Houston we have a problem"
+  const body = `Hi there,\n\nI received a ${code} error page while I was trying to...`
+  return encodeURI(`mailto:${supportEmail}?subject=${subject}&amp;body=${body}`)
+}
+
+const HOME_HREF = "/app/home"
+
+export type ErrorPageProps = {
+  code: number
```

Comment:
Could we use `400 | 404 | ... | number` as the type here?

Reply:
Typescript will squash the type down to just number if we do that, so we'd lose any benefit of listing them.

Reply:
I don't know of a way to get the type to list the statuses that we DO cover but also allow any number as well.

Perhaps we can make the statuses template literals but also allow numbers as well?

So `400` | `422` ... | number

Reply:
Yeah I think it's important they get auto complete. Do you have any ideas here @HeartSquared? Otherwise I'm fine with the string approach.

Reply:
![image](https://github.com/cultureamp/kaizen-design-system/assets/25891850/60932113-01e2-4017-a8cf-3a24ea920a97)

I just tested whether it would work if I made a `NumberSuggestions` and sadly it doesn't. Typing the quotes is what triggers the auto-complete, and braces do not (which is what numbers use).

You could definitely use `StringSuggestions` and therefore accept only strings, but if you want to accept any number, there's no way to offer suggestions (`400 | number` will end up just as `number`).

The other option is to have your type be a subset of numbers and accept a different type (eg. string) as the flexible type. But accepting multiple types is more gross imo so I wouldn't recommend it.

Reply:
@lisatassone is it likely that there will be a custom one? I'm afraid of optimising for an edge case.

I can only think of returning to the `BaseErrorPage` to accept any number, essentially boiling it down to the `ErrorPage` being a strict wrapper with context and passing it through.

```
const ErrorPage = ({ code }) => {
  const content = useErrorMessages(code)
  <BaseErrorPage code={code} {...content} />
}
```

Which doesn't look too bad?

Reply:
@gyfchong I'm not sure. It doesn't have to be a number that's passed so if you want to do what @HeartSquared is suggesting I can add that even though the signature is a bit weird, or we just allow the statuses we cover.

If you can let me know which option you want before midday that would be great. Otherwise feel free to jump in and edit what is there.

Reply:
Let's just allow the statuses we cover for now.